### PR TITLE
[memprof][NFC] Free symbolizer memory eagerly

### PR DESCRIPTION
**Pull request by teresajohnson**
Move the ownership of the symbolizer into symbolizeAndFilterStackFrames
so that it is freed on exit, when we are done with it, to reduce peak
memory in the reader. This reduces about 9G from the peak for one large
profile.

### CODE REVIEW

**Comment by teresajohnson** [comment]
Fixed version of PR75119, which got very messed up after some kind of bad merge.

File: llvm/lib/ProfileData/RawMemProfReader.cpp [hunk 1]
```diff
@@ -344,7 +344,7 @@ Error RawMemProfReader::initialize(std::unique_ptr<MemoryBuffer> DataBuffer) {
       Object, std::move(Context), /*UntagAddresses=*/false);
   if (!SOFOr)
     return report(SOFOr.takeError(), FileName);
-  Symbolizer = std::move(SOFOr.get());
+  auto Symbolizer = std::move(SOFOr.get());
```
Comment (snehasish) [inline llvm/lib/ProfileData/RawMemProfReader.cpp:347]: I wonder if it makes a difference to the peak memory usage if we move L343-L347 after the readRawRawProfile call. That would mean that we initialize the Symbolizer after we free the raw profile data buffer.

**Review by snehasish — APPROVED** [review]
lgtm with an optional idea.

File: llvm/lib/ProfileData/RawMemProfReader.cpp (hunk 1 above)
Comment (teresajohnson) [reply llvm/lib/ProfileData/RawMemProfReader.cpp:347]: Yeah that's a good idea. In fact, we can move it after setupForSymbolization(), although I don't think there is a memory savings from doing that.

**Pull request merged**